test/server/guarddog_impl_test.cc

```diff
@@ -686,8 +692,14 @@ TEST_P(GuardDogActionsTest, MegaMissShouldOnlyReportRelevantThreads) {
   // This will reset the loop interval timer, and should help us
   // synchronize with the guard dog.
   guard_dog_->forceCheckForTest();
+  // Touch the second_dog in case we overslept in the real time system
```

Comment:
should we do this extra petting only when this test is run in real-time mode?

Reply:
We can do that if you really want, but the extra petting it shouldn't affect the simulated time test much as we're really just trying to get one of the watch dogs petted and have the other not petted over the given threshold, so that when we do the final check the latter one triggers the event. 

Petting the former more often doesn't change that.

Reply:
The reason I suggest that is that part of the goal of the simulated time test is that it should be completely deterministic in its behavior regardless of how loaded the testing machine is.

I think this workaround that's specific to real-time means we don't gain the confidence that the simtime version is 100% deterministic.

Reply:
Done.